plugins/inputs/jolokia2/jolokia.go

```diff
@@ -0,0 +1,220 @@
+		SSLCA:              j.SSLCA,
+		SSLCert:            j.SSLCert,
+		SSLKey:             j.SSLKey,
+		InsecureSkipVerify: j.InsecureSkipVerify,
+	})
+}
+
+type JolokiaProxy struct {
+	DefaultFieldPrefix    string
+	DefaultFieldSeparator string
+	DefaultTagPrefix      string
+
+	URL                   string `toml:"url"`
+	DefaultTargetPassword string
+	DefaultTargetUsername string
+	Targets               []JolokiaProxyTargetConfig `toml:"target"`
+
+	Username           string
+	Password           string
+	SSLCA              string `toml:"ssl_ca"`
+	SSLCert            string `toml:"ssl_cert"`
+	SSLKey             string `toml:"ssl_key"`
+	InsecureSkipVerify bool
+	ResponseTimeout    time.Duration `toml:"response_timeout"`
+
+	Metrics []MetricConfig `toml:"metric"`
+}
+
+type JolokiaProxyTargetConfig struct {
+	URL      string `toml:"url"`
+	Username string
+	Password string
+}
+
+func (jp *JolokiaProxy) SampleConfig() string {
+	return `
+  # default_tag_prefix      = ""
+  # default_field_prefix    = ""
+  # default_field_separator = "."
+
+  ## Proxy agent
+  url = "http://localhost:8080/jolokia"
+  # username = ""
+  # password = ""
+  # response_timeout = "5s"
+
+  ## Optional SSL config
+  # ssl_ca   = "/var/private/ca.pem"
+  # ssl_cert = "/var/private/client.pem"
+  # ssl_key  = "/var/private/client-key.pem"
+  # insecure_skip_verify = false
+
+  ## Add targets to query
+  # default_target_username = ""
+  # default_target_password = ""
+  [[inputs.jolokia_proxy.target]]
+    url = "service:jmx:rmi:///jndi/rmi://targethost:9999/jmxrmi"
+	# username = ""
+	# password = ""
+
+  ## Add metrics to read
+  [[inputs.jolokia_proxy.metric]]
+    name  = "java_runtime"
+    mbean = "java.lang:type=Runtime"
+    paths = ["Uptime"]
+`
+}
+
+func (jc *JolokiaProxy) Description() string {
+	return "Read JMX metrics from a Jolokia proxy REST endpoint"
+}
+
+func (jp *JolokiaProxy) buildProxy() (*Agent, error) {
+	proxyConfig := &ProxyConfig{
+		DefaultTargetUsername: jp.DefaultTargetUsername,
+		DefaultTargetPassword: jp.DefaultTargetPassword,
+	}
+
+	for _, target := range jp.Targets {
+		proxyConfig.Targets = append(proxyConfig.Targets, ProxyTargetConfig{
+			URL:      target.URL,
+			Username: target.Username,
+			Password: target.Password,
+		})
+	}
+
+	return NewAgent(jp.URL, &AgentConfig{
+		Username:           jp.Username,
+		Password:           jp.Password,
+		ResponseTimeout:    jp.ResponseTimeout,
+		SSLCA:              jp.SSLCA,
+		SSLCert:            jp.SSLCert,
+		SSLKey:             jp.SSLKey,
+		InsecureSkipVerify: jp.InsecureSkipVerify,
+		ProxyConfig:        proxyConfig,
+	})
+}
+
+func (jc *JolokiaProxy) Gather(acc telegraf.Accumulator) error {
+	var metrics []Metric
+
+	for _, config := range jc.Metrics {
+		metrics = append(metrics, NewMetric(config,
+			jc.DefaultFieldPrefix, jc.DefaultFieldSeparator, jc.DefaultTagPrefix))
+	}
+
+	gatherer := NewGatherer(acc, metrics)
+	proxy, err := jc.buildProxy()
```

Comment:
Building the proxy and agent is also done every gather when it could be done once.  This will create a new transport each Gather which has the potential to cause problems.  Also needs change in Jolokia struct